agent/consul/gateways/gateway_meta_test.go

```diff
@@ -0,0 +1,413 @@
+package gateways
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/consul/agent/consul/state"
+	"github.com/hashicorp/consul/agent/structs"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGetGatewayMeta(t *testing.T) {
```

Comment:
Since this is the only thing that invokes `getGatewayMeta` think we can 💧 it too.